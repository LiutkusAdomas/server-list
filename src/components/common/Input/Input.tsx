import { ChangeEventHandler } from 'react';

type InputPropsType = {
  label: string;
  control: string;
  type: string;
  required: boolean;
  autocomplete: string;
  onChange: ChangeEventHandler<HTMLInputElement>;
  errorMesage?: string;
};

const Input: React.FC<InputPropsType> = ({
  label,
  control,
  type,
  required,
  autocomplete,
  onChange,
  errorMesage,
}) => {
  return (
    <div className="relative">
      <input
        name={control}
        type={type}
        required={required}
        autoComplete={autocomplete}
        className={`appearance-none relative block w-full px-3 py-2 my-4 border placeholder-gray-500 text-gray-900 rounded-lg shadow-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm ${
          errorMesage ? 'border-red-500 mb-6' : 'border-gray-300'
        }`}
        placeholder={label}
        onChange={onChange}
      />
      {errorMesage && (
        <span className="absolute -bottom-5 text-xs text-red-500">{errorMesage}</span>
      )}
    </div>
  );
};

export default Input;
