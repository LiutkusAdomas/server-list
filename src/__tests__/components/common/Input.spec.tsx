import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import { Input } from '../../../components/common/Input/Input';

describe('Input component', () => {
  it('should render an input element without error', () => {
    const { asFragment } = render(
      <Input
        label="Test"
        control="test"
        type="text"
        required={true}
        autocomplete="username"
        onChange={vi.fn()}
      />
    );
    expect(asFragment).toMatchSnapshot();
  });
  it('should render an input element with an error', () => {
    const { asFragment } = render(
      <Input
        label="Test"
        control="test"
        type="text"
        required={true}
        autocomplete="username"
        onChange={vi.fn()}
        errorMesage="This is an error message"
      />
    );
    expect(asFragment).toMatchSnapshot();
  });
});
