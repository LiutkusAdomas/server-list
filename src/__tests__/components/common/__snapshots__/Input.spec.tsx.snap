// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input component > should render an input element with an error 1`] = `[Function]`;

exports[`Input component > should render an input element without error 1`] = `[Function]`;
